### PR TITLE
3 create a dem from many laz tiles

### PR DESCRIPTION
**Pull request by rosepearson**
Expand the pipeline to support many tiles from a single LiDAR dataset located on OpenTopography. Each tile is treated separately and added individually to the resulting dem.

### CODE REVIEW

**Comment by rosepearson** [comment]
Work that still needs to be done as part of this PR:
* Add a linter to check spelling and formatting as part of the commit/pushing process using a hook
* Once finished with changes update the test benchmarks
*

**Review by LukeParky — CHANGES_REQUESTED** [review]
Good improvements in clarity included, and easy to follow structure of source code with the tiling additions

File: src/geofabrics/processor.py [hunk 1]
```diff
@@ -15,110 +12,117 @@
...
+        # Note correctly only consider one LiDAR dataset.
         lidar_dataset_index = 0
-        lidar_tile_index = 0
-        
-        ### instruction values and other set values
+        area_threshold = 10.0/100  # 10%
+
+        # load in instruction values or set to defaults
+        verbose = self.instructions['instructions']['instructions']['verbose'] if 'verbose' in \
+            self.instructions['instructions']['instructions'] else True
```
[the remark this answers is not in the document]
Comment (rosepearson) [reply src/geofabrics/processor.py:41]: Yes - it's been bothering me too! I'll sort something out!

[the remark this answers is not in the document]
Comment (rosepearson) [reply src/geofabrics/processor.py]: Great. Thanks @LukeParky - changes made.

File: src/geofabrics/processor.py [hunk 2]
```diff
@@ -15,110 +12,150 @@
...
+
         self.catchment_geometry = None
         self.dense_dem = None
         self.reference_dem = None
         self.bathy_contours = None
         self.result_dem = None
-        
+
+    def get_instruction_path(self, key: str) -> str:
+        """ Return the file path from the instruction file. Raise an erro if not in the instructions. """
```
Comment (kinow) [inline src/geofabrics/processor.py:33]: s/erro/error

File: src/geofabrics/processor.py [hunk 3]
```diff
@@ -15,110 +12,150 @@
...
+
+        assert key in self.instructions['instructions']['data_paths'], "Key missing from data paths"
+        return self.instructions['instructions']['data_paths'][key]
+
+    def get_resolution(self) -> float:
+        """ Return the resolution from the instruction file. Raise an erro if not in the instructions. """
```
Comment (kinow) [inline src/geofabrics/processor.py:39]: s/erro/error

**Comment by rosepearson** [comment]
Thanks for the review @LukeParky and @kinow. Let me know if there is anything else you'd like addressed.

**Review by LukeParky — APPROVED** [review]
The new changes to instructions are much more readable in the main body of the code. Looks good to pass PR

**Pull request merged**